plugins/call_button/call_button.tsx

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {CSSProperties, useState, useEffect, useRef} from 'react';
+import {injectIntl, IntlShape} from 'react-intl';
+import classNames from 'classnames';
+
+import MenuWrapper from 'components/widgets/menu/menu_wrapper';
+import Menu from 'components/widgets/menu/menu';
+import {Constants} from 'utils/constants';
+
+import {Channel, ChannelMembership} from 'mattermost-redux/types/channels';
+import {PluginComponent} from 'types/store/plugins';
+
+import './call_button.scss';
+
+type Props = {
+    currentChannel: Channel;
+    channelMember?: ChannelMembership;
+    intl: IntlShape;
+    locale: string;
+    pluginCallComponents: PluginComponent[];
+    sidebarOpen: boolean;
+}
+
+function CallButton(props: Props) {
+    if (props.pluginCallComponents.length === 0) {
+        return null;
+    }
+
+    const {formatMessage} = props.intl;
+    const [active, setActive] = useState(false);
+    const [clickEnabled, setClickEnabled] = useState(true);
+    const prevSidebarOpen = useRef(props.sidebarOpen);
+
+    useEffect(() => {
+        if (prevSidebarOpen.current && !props.sidebarOpen) {
+            setClickEnabled(false);
+            setTimeout(() => {
+                setClickEnabled(true);
+            }, Constants.CHANNEL_HEADER_BUTTON_DISABLE_TIMEOUT);
+        }
+        prevSidebarOpen.current = props.sidebarOpen;
+    }, [props.sidebarOpen]);
+
+    const style = {
+        container: {
+            marginTop: '16px',
+            height: '32px',
```

Comment:
I am not 100% sure, but you can just pass in `int`s as values, since the standard is to use `px`

```suggestion
            marginTop: 16,
            height: 32,
```

Reply:
just tested and it does work as expected